spec/index.bs

```diff
@@ -1493,6 +1497,103 @@ interface GPUTextureUsage {
 };
 </script>
 
+### <dfn method for=GPUDevice>createTexture(descriptor)</dfn> ### {#GPUDevice-createTexture}
+
+<div algorithm="GPUDevice.createTexture">
+  **Arguments:**
+    - {{GPUTextureDescriptor}} |descriptor|
+
+  **Returns:** {{GPUTexture}}
+
+  1. If device is lost, or if this call doesn't follow the [$createTexture Valid Usage$], return an error texture.
+  1. Let |t| be a new {{GPUTexture}} object.
+  1. Set |t|.{{GPUTexture/[[textureSize]]}} to |descriptor|.{{GPUTextureDescriptor/size}}.
+  1. Set |t|.{{GPUTexture/[[sampleCount]]}} to |descriptor|.{{GPUTextureDescriptor/sampleCount}}.
+  1. Set |t|.{{GPUTexture/[[dimension]]}} to |descriptor|.{{GPUTextureDescriptor/dimension}}.
+  1. Set |t|.{{GPUTexture/[[format]]}} to |descriptor|.{{GPUTextureDescriptor/format}}.
+  1. Set |t|.{{GPUTexture/[[textureUsage]]}} to |descriptor|.{{GPUTextureDescriptor/usage}}.
+  1. Return |t|.
+
+  <div algorithm class=validusage>
+    <dfn abstract-op>maximum valid mipLevel</dfn>
+      Given a {{GPUTextureDescriptor/dimension}} and a {{GPUTextureDescriptor/size}},
+
+      1. Calculate the values of |w|, |h|, and |d|. If the {{GPUTextureDescriptor/dimension}} is "1d":
+
+        1. Let |w| = {{GPUTextureDescriptor/size}}.[=Extent3D/width=].
+        1. Let |h| = 1.
+        1. Let |d| = 1.
+
+      1. Else if the {{GPUTextureDescriptor/dimension}} is "2d":
+
+        1. Let |w| = {{GPUTextureDescriptor/size}}.[=Extent3D/width=].
+        1. Let |h| = {{GPUTextureDescriptor/size}}.[=Extent3D/height=].
+        1. Let |d| = 1.
+
+      1. Else (the {{GPUTextureDescriptor/dimension}} is "3d"):
+
+        1. Let |w| = {{GPUTextureDescriptor/size}}.[=Extent3D/width=].
+        1. Let |h| = {{GPUTextureDescriptor/size}}.[=Extent3D/height=].
+        1. Let |d| = {{GPUTextureDescriptor/size}}.[=Extent3D/depth=].
+
+      1. Let |m| = the maximum value of |w|, |h|, and |d|.
+      1. Return one plus the greatest integral value of |x| for which 2^|x| <= |m|.
+
+    </dfn>
+  </div>
+
+  <div algorithm class=validusage>
+    <dfn abstract-op>createTexture Valid Usage</dfn>
+      Given a {{GPUDevice}} |this| and a {{GPUTextureDescriptor}} |descriptor|
+      the following validation rules apply:
+
+      1. |this| must be a [=valid=] {{GPUDevice}}.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be nonzero.
+      1. If |descriptor|.{{GPUTextureDescriptor/dimension}} is "1d":
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 2048.
```

Comment:
Vulkan has a minimum number at 256 and a large number of Android devices expose that value. Ditto for the number of array layers for other dimensions.

Reply:
Resolved in latest update